hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationMonitor.java

```diff
@@ -73,6 +76,7 @@
     private final ILogger logger;
     private final ScheduledExecutorService scheduler;
     private final Address thisAddress;
+    private final ConcurrentMap<Address, AtomicLong> lastHeartbeatPerMember = new ConcurrentHashMap<Address, AtomicLong>();
```

Comment:
Also other threads; any thread can make an investigation regarding the last heartbeat of that given member (for the invocationfuture error message for example).